pkg/decision/composite_service.go

```diff
@@ -35,16 +35,34 @@ type CompositeService struct {
 	notificationCenter         notification.Center
 }
 
+// CSOptionFunc allows customization of the CompositeService returned from NewCompositeService
+type CSOptionFunc func(*CompositeService)
+
+// WithExperimentOverrides applies the argument experiment overrides to a composite service
+func WithExperimentOverrides(experimentOverrides map[OverrideKey]string) CSOptionFunc {
```

Comment:
Outside the scope of this PR, but I think it's a little cumbersome to have to add the option func in these two places. I've been thinking about simplifying the decision package a little by having this `composite_service` encompass the `[]ExperimentService` and `[]FeatureService` from the `composite_experiment_service` and `composite_feature_service` structs and get rid of those two classes entirely since those are just wrappers for the arrays and looping logic. What do you think?